### PR TITLE
Feature/fix transient filter

### PR DESCRIPTION
**Pull request by roesslerj**

### CODE REVIEW

File: src/main/java/de/retest/recheck/ignore/SearchFilterFiles.java [hunk 1]
```diff
@@ -83,7 +84,8 @@ private SearchFilterFiles() {}
 	private static List<Pair<String, FilterLoader>> loadFiltersFromDirectory( final Path directory ) {
 		try ( final Stream<Path> paths = Files.walk( directory ) ) {
 			return paths.filter( Files::isRegularFile ) //
-					.filter( file -> file.toString().endsWith( FILTER_EXTENSION ) ) //
+					.filter( file -> file.toString().endsWith( FILTER_EXTENSION )
+							|| file.toString().endsWith( FILTER_JS_EXTENSION ) ) //
```
Comment (diba1013) [inline src/main/java/de/retest/recheck/ignore/SearchFilterFiles.java]: Extract this into a method?

File: src/main/java/de/retest/recheck/RecheckOptions.java [hunk 2]
```diff
@@ -88,10 +90,11 @@ private File getSuitePath() {
...
 		private String suiteName = null;
 		private boolean reportUploadEnabled = false;
-		private Filter filter = null;
-		private final List<Filter> filterToAdd = new ArrayList<>();
+		private Filter ignoreFilter = null;
+		private final List<Filter> ignoreFilterToAdd = new ArrayList<>();
 
-		private RecheckOptionsBuilder() {}
+		private RecheckOptionsBuilder() {
+		}
```
Comment (diba1013) [inline src/main/java/de/retest/recheck/RecheckOptions.java]: I prefer the old compact brackets...

**Review by diba1013 — CHANGES_REQUESTED** [review]

File: src/main/java/de/retest/recheck/ignore/SearchFilterFiles.java [hunk 3]
```diff
@@ -121,4 +123,18 @@ private SearchFilterFiles() {}
 	private static String getFileName( final Path path ) {
 		return path.getFileName().toString();
 	}
+
+	public static Filter getFilterByName( final String name ) {
+		final Filter result = toFileNameFilterMapping().get( name );
```
[the remark this answers is not in the document]
Comment (roesslerj) [reply src/main/java/de/retest/recheck/ignore/SearchFilterFiles.java:132]: Totally agree, but will be put into separate PR and task...

File: src/main/java/de/retest/recheck/ignore/SearchFilterFiles.java [hunk 4]
```diff
@@ -121,4 +127,18 @@ private SearchFilterFiles() {}
 	private static String getFileName( final Path path ) {
 		return path.getFileName().toString();
 	}
+
+	public static Filter getFilterByName( final String name ) {
```
Comment (diba1013) [inline src/main/java/de/retest/recheck/ignore/SearchFilterFiles.java:131]: Oh, just realized. Do you want to create two quick and simple tests, that are not integration tests?

**Review by diba1013 — CHANGES_REQUESTED** [review]

File: src/main/java/de/retest/recheck/ignore/SearchFilterFiles.java (hunk 4 above)
Comment (roesslerj) [reply src/main/java/de/retest/recheck/ignore/SearchFilterFiles.java:131]: To test what? I adapted one of the existing tests, to make sure .js files are included: https://github.com/retest/recheck/pull/394/commits/22d1673d57b70306ebbbb7c042a306c9df5acc7f

Comment (diba1013) [reply src/main/java/de/retest/recheck/ignore/SearchFilterFiles.java:131]: Like explicitly test this method that it finds a filter or throws an exception. The only tests are when `RecheckOptions` are created.

Comment (roesslerj) [reply src/main/java/de/retest/recheck/ignore/SearchFilterFiles.java:131]: Done

**Review by diba1013 — APPROVED** [review]

**Pull request merged**